pkg/operator/ceph/object/objectstore.go

```diff
@@ -757,15 +759,48 @@ func enableRGWDashboard(context *Context) error {
 	if err != nil || errCode != 0 {
 		return errors.Wrapf(err, "failed to create user %q", DashboardUser)
 	}
-	args := []string{"dashboard", "set-rgw-api-access-key", *u.AccessKey}
-	cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
-	_, err = cephCmd.Run()
-	if err != nil {
-		return errors.Wrapf(err, "failed to set user %q accesskey", DashboardUser)
+
+	cephVersion := mgr.FileBasedPasswordSupported(context.clusterInfo)
+	var args []string
+	var secretFile *os.File
+	// for latest Ceph versions
+	if cephVersion {
+		file, err := mgr.CreateTempPassFile(*u.AccessKey)
+		if err != nil {
+			return err
+		}
+
+		args = []string{"dashboard", "set-rgw-api-access-key", "-i", file.Name()}
+		cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
```

Comment:
How about also factoring out this call to `NewCephCommand()`? This just means we would have different vars for `file` and `args` for each of the two commands.

Reply:
done.